time_formatter.rb

```diff
@@ -0,0 +1,24 @@
+# contain format logic
+class TimeFormatter
+  AVAILABLE = %w[year month day hour minute second].freeze
+
+  def initialize(input)
+    @input, @unknown = input.partition { |v| AVAILABLE.member?(v) }
+  end
+
+  def perform
+    valid? ? divided_time : @unknown
+  end
+
+  def divided_time
+    formath = (AVAILABLE & @input).map do |m|
```

Comment:
Зачем еще раз пересечение находить? В input уже только известные форматы. Тут скорее проблема нейминга, из-за которой ты сам запутался что у тебя в переменной
```suggestion
        format = @input.map do |m|
```